infra/terraform/modules/data_access/memberships/memberships.py

```diff
@@ -0,0 +1,100 @@
+
+'''
+Environment variables:
+ - STAGE, e.g. "dev", "alpha", etc...
+ - IAM_ARN_BASE, e.g. "arn:aws:iam::1234"
+'''
+
+
+import json
+import os
+
+import boto3
+import botocore.exceptions
+
+POLICY_READ_ONLY = "readonly"
+POLICY_READ_WRITE = "readwrite"
+
+
+class InvalidPolicyType(Exception):
+    pass
+
+
+def attach_bucket_policy(event, context):
+    """
+    Attaches the team bucket IAM policy to the user's IAM role
+
+    event = {
+        "user": {"username": "alice"},
+        "team": {"slug": "justice-league"},
+        "policy": {"type": "readwrite"},
+    }
+    """
+    policy_type = event["policy"]["type"]
+    validate_policy_type(policy_type)
+
+    username = event["user"]["username"]
+    team_slug = event["team"]["slug"]
+
+    client = boto3.client("iam")
+    client.attach_role_policy(
+        RoleName=role_name(username),
+        PolicyArn=policy_arn(team_slug, policy_type),
+    )
+
+
+def detach_bucket_policies(event, context):
+    """
+    Detaches the team bucket IAM policies from the user's IAM role
+
+    event = {
+        "user": {"username": "alice"},
+        "team": {"slug": "justice-league"}
+    }
+    """
+    username = event["user"]["username"]
+    team_slug = event["team"]["slug"]
+    name = role_name(username)
+
+    client = boto3.client("iam")
+    for policy_type in [POLICY_READ_WRITE, POLICY_READ_ONLY]:
+        # Be sure we detach all policies without stopping early
+        try:
+            client.detach_role_policy(
+                RoleName=name,
+                PolicyArn=policy_arn(team_slug, policy_type),
+            )
+        except botocore.exceptions.ClientError:
+            pass
```

Comment:
Well in this specific case the error is when we try to detach one of the two policies but the policy is not attached (e.g. role has only readonly policy and we try to detach readwrite).

As the alternative would be to check if the policy is attached then remove it but there is no harm in trying to detach something which is not attached. We save some logic and an additional request all the times. We may log it but it may sound alarming/cause noise without being actually useful.